design/yugabytedb-rook-design.md

```diff
@@ -0,0 +1,85 @@
+YugaByte DB is a high-performance distributed transactional database. (More information [here](https://docs.yugabyte.com/latest/introduction/)). This issue outlines design and interfaces for a Rook YugaByte operator for running YugaByte on Kubernetes with Rook as the underlying storage engine.
+
+**What is use case behind this feature:**
+
+A cluster of YugaByte DB backed by Rook should be created using the `clusters.yugabytedb.rook.io` custom resource definition. Below is a sample custom resource spec for creating a 3-master, 3 t-servers cluster using the CRD. The sample is followed by an explanation of different configuration options available on the YugaByte DB CRD.
+
+## Sample
+
+```yaml
+apiVersion: yugabytedb.rook.io/v1alpha1
+kind: Cluster
+metadata:
+  name: rook-yugabytedb
+  namespace: rook-yugabytedb
+spec:
```

Comment:
What if `master` and `tserver` were top-level elements of the spec? It just depends if users will think more intuitively about network, replicas, and scope as the top-level items, or if master and tserver are the most important. For example, the schema might look like:
```yaml
  master:
    replicas: 3
    ports:
       ui: 7000
       grpc: 7100
    volumeClaimTemplates:
    - metadata:
        name: datadir
      spec:
        accessModes: [ "ReadWriteOnce" ]
        resources:
          requests:
            storage: 1Gi
        storageClassName: standard
  tserver:
    replicas: 3
    ports:
       ui: 9000
       grpc: 9100
       cassandra: 9042
       redis: 6379
       postgres: 5433
    volumeClaimTemplates:
    - metadata:
        name: datadir
      spec:
        accessModes: [ "ReadWriteOnce" ]
        resources:
          requests:
            storage: 1Gi
        storageClassName: standard
```
I'm fine with your original schema, just something for you to consider.

Reply:
Thanks - this is a great suggestion and I really liked this structure better. It is much more intuitive to user too.